cmd/simulator/simulate/loader/upkeep.go

```diff
@@ -0,0 +1,72 @@
+package loader
+
+import (
+	"sync"
+
+	"github.com/smartcontractkit/ocr2keepers/cmd/simulator/config"
+	"github.com/smartcontractkit/ocr2keepers/cmd/simulator/simulate/chain"
+)
+
+const (
+	CreateUpkeepNamespace = "Emitting create upkeep transactions"
+	emitLogNamespace      = "Emitting log events"
+)
+
+// UpkeepConfigLoader provides upkeep configurations to a block broadcaster. Use
+// this loader to introduce upkeeps or change upkeep configs at specific block
+// numbers.
+type UpkeepConfigLoader struct {
+	// provided dependencies
+	progress ProgressTelemetry
+
+	// internal state values
+	mu     sync.RWMutex
+	create map[string][]interface{}
+}
+
+// NewUpkeepConfigLoader ...
+func NewUpkeepConfigLoader(plan config.SimulationPlan, progress ProgressTelemetry) (*UpkeepConfigLoader, error) {
+	// combine all upkeeps together for transmit
+	allUpkeeps, err := chain.GenerateAllUpkeeps(plan)
+	if err != nil {
+		return nil, err
+	}
+
+	create := make(map[string][]interface{})
+	for _, upkeep := range allUpkeeps {
+		evts, ok := create[upkeep.CreateInBlock.String()]
+		if !ok {
+			evts = []interface{}{}
+		}
+
+		create[upkeep.CreateInBlock.String()] = append(evts, chain.UpkeepCreatedTransaction{
+			Upkeep: upkeep,
+		})
+	}
+
+	if progress != nil {
```

Comment:
In what cases would progress be nil? Would it be worth even creating a no-op implementation we could default to to prevent nil checks like this?

Reply:
That was just for testing since I didn't want to create a telemetry object for every unit test. I can probably come up with some sort of no-op to set the value in the constructor to remove later nil checks on that value.

Reply:
You could just create a mock with an empty `RegisterFn`?